Cython/Compiler/Main.py

```diff
@@ -653,8 +672,6 @@ def compile(source, options = None, full_module_name = None, **kwds):
     """
     options = CompilationOptions(defaults = options, **kwds)
     if isinstance(source, str):
-        if not options.timestamps:
-            return compile_single(source, options, full_module_name)
```

Comment:
This change is causing test_grammar.py tests failing. Not sure what is best to do here. Should be compile_single extended to support cache (context creation needs to be added then)?

Reply:
The cache is persistent, so it shouldn't depend on the number of modules that users compile whether it's being used or not.

Reply:
Reverted back to original logic